python/dgl/dataloading/base.py

```diff
@@ -271,9 +410,171 @@ def sample(self, g, seed_edges):
             g, seed_edges, exclude, self.reverse_eids, self.reverse_etypes,
             self.output_device)
 
-        input_nodes, _, blocks = self.block_sampler.sample_blocks(g, seed_nodes, exclude_eids)
+        input_nodes, _, blocks = self.sampler.sample(g, seed_nodes, exclude_eids)
 
         if self.negative_sampler is None:
             return self.assign_lazy_features((input_nodes, pair_graph, blocks))
         else:
             return self.assign_lazy_features((input_nodes, pair_graph, neg_graph, blocks))
+
+def as_edge_prediction_sampler(
+        sampler, exclude=None, reverse_eids=None, reverse_etypes=None, negative_sampler=None,
+        prefetch_labels=None):
+    """Wrap a sampler for node classification/regression into another sampler for
+    edge classification/regression or link prediction.
```

Comment:
```python
"""Create an edge-wise sampler from a node-wise sampler.

For each batch of edges, the sampler applies the provided node-wise sampler to their source
and destination nodes to get the subgraph samples. It also generates negative edges if a
negative sampler is provided.
"""
```

Reply:
<img width="486" alt="two_tower_edge" src="https://user-images.githubusercontent.com/4057701/155685352-9725f6da-4917-4c1f-9512-f56304d1e1ec.png">

Consider having this image in the API doc.